### ISSUE TITLE
async function support 

### ISSUE THREAD

**Issue by zaaack** [jayphelps/core-decorators.js#100]
Throttle/debounce won't work for async functions, hope could work with functions return a promise.  Idea: https://github.com/sindresorhus/promise-fun

**Comment by BurtHarris** [comment]
Certainly makes sense to me, but both decorator are currently deprecated from this package, and will be removed soon.

**Comment by jayphelps** [comment]
Indeed, but let's keep it open because this is still valid in general for all decorators. I haven't had time to look into why things are different.
